Mathlib/Topology/Algebra/LinearTopology.lean

```diff
@@ -0,0 +1,133 @@
+/-
+Copyright (c) 2024 Antoine Chambert-Loir, María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Antoine Chambert-Loir, María Inés de Frutos-Fernández, Anatole Dedecker
+-/
+
+import Mathlib.RingTheory.TwoSidedIdeal.Operations
+import Mathlib.Topology.Algebra.Ring.Basic
+import Mathlib.Topology.Algebra.OpenSubgroup
+
+/-! # Linear topologies on rings
+
+Following Bourbaki, *Algebra II*, chapter 4, §2, n° 3, a topology on a ring `R` is *linear* if
+it is invariant by translation and admits a basis of neighborhoods of 0 consisting of
+two-sided ideals.
+
+- `tendsto_mul_zero_of_left`: for `f, g : ι → R` such that `f i` converges to `0`,
+`f i * g i` converges to `0`.
+
+- `tendsto_mul_zero_of_right`: for `f, g : ι → R` such that `g i` converges to `0`,
+`f i * g i` converges to `0`.
+
+## Instances
+
+- A discrete topology is a linear topology
+
+## Note on the implementation
+
+The definition of Bourbaki doesn't presuppose, but implies, that a linear topology on a ring `R` is
+a ring topology. However, in some of our lemmas, we already assume that `R` is a topological ring.
+This unnecessary assumption will be made unnecessary by results in the ongoing PR #18437.
+Anyway, the idea will be to first define a topology on `R`, and then
+prove that it makes `R` a topological ring, and that it is a linear topology.
+
+-/
+
+open scoped Topology
+open Filter
+
+namespace IsLinearTopology
+
+section Ring
+
+/-- A topology on a ring is linear if its topology is defined by a family of ideals. -/
+class _root_.IsLinearTopology (R : Type*) [Ring R] [TopologicalSpace R] where
```

Comment:
I know nothing about this, but [Wikipedia](https://en.wikipedia.org/wiki/Linear_topology) defines linear topology in a different context, and it doesn't agree with this definition (it uses left submodules, not two sided ideals). I do not suggest that you change the definition (see start of this comment), but I think that existence of different definitions and some relations between them should be discussed in the module docstring.